src/main/java/org/terasology/climateConditions/ExtremeClimateConditionSystem.java

```diff
@@ -0,0 +1,198 @@
+    @ReceiveEvent(components = {PlayerCharacterComponent.class, CharacterMovementComponent.class})
+    public void ChangeSnowEffect(MovedEvent event, EntityRef player, LocationComponent location, CharacterMovementComponent movement) {
+        float height = event.getPosition().getY();
+        float deltaHeight = event.getDelta().getY();
+        float lastHeight = height - deltaHeight;
+        if(height > thresholdHeight && lastHeight <= thresholdHeight) {
+            delayManager.addPeriodicAction(player, FROSTBITE_DAMAGE_ACTION_ID, 0, healthDecreaseInterval);
+            player.send(new ReduceSpeedEvent());
+
+        }
+
+        if(height < thresholdHeight && lastHeight >= thresholdHeight) {
+            delayManager.cancelPeriodicAction(player, FROSTBITE_DAMAGE_ACTION_ID);
+            player.send(new ChangeSpeedToDefaultEvent());
+
+            if (player.hasComponent(SnowParticleComponent.class)) {
+                EntityRef particleEntity = player.getComponent(SnowParticleComponent.class).particleEntity;
+                if (particleEntity != EntityRef.NULL) {
+                    particleEntity.destroy();
+                }
+                player.removeComponent(SnowParticleComponent.class);
+            }
+        }
+    }
+
+    @ReceiveEvent(components = {LocationComponent.class})
+    public void onPeriodicActionTriggered(PeriodicActionTriggeredEvent event, EntityRef unusedEntity) {
+        if (event.getActionId().equals(FROSTBITE_DAMAGE_ACTION_ID)) {
+            for (EntityRef entity : entityManager.getEntitiesWith(LocationComponent.class, AliveCharacterComponent.class)) {
+                    // Check to see if health should be decreased
+                    LocationComponent location = entity.getComponent(LocationComponent.class);
+                    float height = location.getLocalPosition().getY();
+                    //   float deltaHeight = location.getLastPosition().getY();
+                    // float lastHeight = height - deltaHeight;
+                    if (height > thresholdHeight) {
+                        Prefab frostbiteDamagePrefab = prefabManager.getPrefab("ClimateConditions:FrostbiteDamage");
+                        entity.send(new DoDamageEvent(healthDecreaseAmount, frostbiteDamagePrefab));
+
+                            EntityRef particleEntity = entityManager.create("climateConditions:snowParticleEffect");
+                            LocationComponent targetLoc = entity.getComponent(LocationComponent.class);
+                            LocationComponent childLoc = particleEntity.getComponent(LocationComponent.class);
+                            childLoc.setWorldPosition(targetLoc.getWorldPosition());
+                            Location.attachChild(entity, particleEntity);
+                            particleEntity.setOwner(entity);
+                            Vector3f direction = targetLoc.getLocalDirection();
+                            direction.normalize();
```

Comment:
I would generally advice to extract code into methods as early as possible. This has several benefits, and the most important one imho is the "readability" of the overall algorithm. I prefer to look at code where I can understand what is happening (on a high level) on the first glance, and drill down into details whenever I need to. 

In pseudo-code,  I would like to see this method look something like this:

```java
/**
 * Apply periodic frostbite effects to the player.
 */
@ReceiveEvent(components = {LocationComponent.class})
public void onPeriodicFrostbite(PeriodicActionTriggeredEvent event, EntityRef player) {
    if (event.getActionId().equals(FROSTBITE_DAMAGE_ACTION_ID)) {
        applyFrostbiteDamage(player);
        updateSnowParticleEffects(player);
    }
}
```